.github/local-actions/branch-manager/lib/main.ts

```diff
@@ -0,0 +1,161 @@
+import {
+  assertValidGithubConfig,
+  getConfig,
+  GithubConfig,
+  setConfig,
+} from '../../../../ng-dev/utils/config.js';
+import {AuthenticatedGitClient} from '../../../../ng-dev/utils/git/authenticated-git-client.js';
+import {
+  ANGULAR_ROBOT,
+  getAuthTokenFor,
+  revokeActiveInstallationToken,
+} from '../../../../github-actions/utils.js';
+import {MergeConflictsFatalError} from '../../../../ng-dev/pr/merge/failures.js';
+import {chdir} from 'process';
+import {spawnSync} from 'child_process';
+
+interface CommmitStatus {
+  state: 'pending' | 'error' | 'failure' | 'success';
+  description: string;
+}
+
+async function main(repo: {owner: string; repo: string}, token: string, pr: number) {
+  if (isNaN(pr)) {
+    core.setFailed('The provided pr value was not a number');
+    return;
+  }
+
+  // Because we want to perform this check in the targetted repository, we first need to check out the repo
+  // and then move to the directory it is cloned into.
+  chdir('/tmp');
+  console.log(
+    spawnSync('git', [
+      'clone',
+      `https://github.com/${repo.owner}/${repo.repo}.git`,
+      './branch-manager-repo',
+    ]).output.toString(),
+  );
+  chdir('/tmp/branch-manager-repo');
+
+  // Manually define the configuration for the pull request and github to prevent having to
+  // checkout the repository before defining the config.
+  // TODO(josephperrott): Load this from the actual repository.
+  setConfig(<{pullRequest: PullRequestConfig; github: GithubConfig}>{
+    github: {
+      mainBranchName: 'main',
+      owner: repo.owner,
+      name: repo.repo,
+    },
+    pullRequest: {
+      githubApiMerge: false,
+    },
+  });
+  /** The configuration used for the ng-dev tooling. */
+  const config = await getConfig([assertValidGithubConfig, assertValidPullRequestConfig]);
+
+  AuthenticatedGitClient.configure(token);
+  /** The git client used to perform actions. */
+  const git = await AuthenticatedGitClient.get();
+
+  /** The pull request after being retrieved and validated. */
+  const pullRequest = await loadAndValidatePullRequest(
+    {git, config},
+    pr,
+    new PullRequestValidationConfig(),
+  );
+  core.info('Validated PR information:');
+  core.info(JSON.stringify(pullRequest));
+  /** Whether any fatal validation failures were discovered. */
+  let hasFatalFailures = false;
+  /** The status information to be pushed as a status to the pull request. */
+  let statusInfo: CommmitStatus = await (async () => {
+    // Log validation failures and check for any fatal failures.
+    if (pullRequest.validationFailures.length !== 0) {
+      core.info(`Found ${pullRequest.validationFailures.length} failing validation(s)`);
+      for (const failure of pullRequest.validationFailures) {
+        hasFatalFailures = !failure.canBeForceIgnored || hasFatalFailures;
+        await core.group('Validation failures', async () => {
```

Comment:
I'm not familar with `.group`. Does this need to be outside the for loop?

Reply:
The `.group` method allows you to execute a callback and occurs between a `.groupStart` and `.groupEnd` call in logging.

I think in situations like this where we are just listing out values, its better readability than the alternative.